setup.py

```diff
@@ -92,6 +92,8 @@
         "dvc[ssh]>=2",
         "matplotlib",
         "scipy",
+        # To enable GPUs with spacy.
+        "spacy[cuda102]",
```

Comment:
If I correctly understand, this will attempt at installing the `extra-requires` of `spacy` for `cuda102`.
So I have a couple of questions related to this.

1. Is this going to work on a system that has no `cuda` installed, or a version of `cuda` different from `10.2`? And if installing `bluesearch[data_and_models]` now requires `cuda` to install, isn't this going to be a pretty big change (not sure we want that)?
2. On Mac OS this should fail because no wheels will be found for this OS, see [here](https://github.com/cupy/cupy/issues/2490#issuecomment-655351955).
3. Maybe we can just put `cuda102` in the `requirements-data_and_models.txt` but not in the `setup.py`?

Reply:
1. Yes it seems so, but you can be more generic by specifying only `spacy[cuda]`.
2. Interesting I didn't know.

Those are really good points! What do you think is the best then: 
- Putting this installation line (`pip install spacy[cuda102]`) in `docker/mining_cache.Dockerfile` so we have it only for the mining cache creation and are sure of the environment - or -
- Keeping only the `cuda102` in the `requirements-data_and_models.txt` and removing it from `setup.py` ?

Reply:
> Yes it seems so, but you can be more generic by specifying only spacy[cuda].

By specifying the generic `[cuda]` you will however compile from source, instead of getting a wheel (much faster). At least, this is what I understand from reading [here](https://spacy.io/usage#gpu).

Also, wouldn't the `[cuda]` still be breaking on any system that doesn't have `cuda` binaries installed?

Reply:
Yes I agree with both points!

Reply:
We are by the way in an interesting situation regarding how we handle the dependencies.

Indeed, the logic to create the mining cache is `not` part of `data_and_models/`. However, it makes sense to us to put the CUDA requirement in the `data_and_models` requirements. This is an inconsistency in how we handle the dependencies.

In the current situation, I am fine with this inconsistency. Otherwise, we would need to tackle as part of the current issue the long running point from #280. We are actually in the case of "pre-computing inference results from the models" as described in #280.

Reply:
This being said, to tackle the current situation, I am for the following suggested solution:

> Maybe we can just put `cuda102` in the `requirements-data_and_models.txt` but not in the `setup.py`?

Reply:
I have removed `spacy[cuda]` from our `setup.py` but kept `cupy-cuda102==8.6.0` into our `requirements-data_and_models.txt` (in af5fa10).